simple/rdm_shared_ctx.c

```diff
@@ -496,10 +498,10 @@ static int run(void)
 		goto out;
 
 	run_test();
-
+	ft_finalize(ep[0], scq, rcq, remote_fi_addr[0]);
```

Comment:
We need to add a local finalize method for scalable ep and shared context since each ep is addressable and we have to make sure we wait on the right ep for the completion. Here, the ft_finalize causes the hang issue.